### ISSUE TITLE
Gotu Entity Clone

### ISSUE THREAD

**Issue by maikmb** [herbsjs/gotu#36]
**Is your feature request related to a problem? Please describe.**
Gotu must have a method `clone` in order to be able to create a copy of an instance.

**Describe the solution you'd like**
This methods can accepts an optional overwrite object that permits you to overwrite some attributes of the copy.

**Additional context**

```
const User = 
    entity('User', {
        name: field(String),
        lastAccess: field(Date),
        accessCount: field(Number),
        features: field([Feature]),
        plan: field(Plan),
    })

const user = new User()
user.name = "Beth"
user.plan.monthlyCost = 10
user.features = [ 
    new Feature(),
    new Feature(),
    new Feature()
]

const cloneUserWithNoOverwrite = user.clone(); // User { name: 'Me }
const cloneWithOverwrite = user.clone({ name: 'Myself' }); // User { name: 'Myself' }
```

**Comment by dalssoft** [comment]
You could use `.fromJSON` to clone.

```js
const { entity, field } = require('gotu')

const Customer =
    entity('Customer', {
        name: field(String)
    })

const customer1 = new Customer()
customer1.name = 'Maik'
const customer2 = Customer.fromJSON(customer1)
```

**Comment by thgpdr** [comment]
Maybe .clone() would be nice for learning curve, but its a **nice to have** and not a **must have**
In a more clean/elegant way, we could also do this

```javascript
const { entity, field } = require('gotu')
const Customer =
    entity('Customer', {
        name: field(String)
    })

const customer1 = new Customer()
customer1.name = 'Lucas'
customer1.age = 20
customer1.gender = 'Male'

const customer2 = Customer.fromJSON({...customer1, name: 'Maik' })
//customer1.name = 'Maik'
//customer1.age = 20
//customer1.gender = 'Male'
```

**Comment by italojs** [comment]
talking abou learning curve, I think this "clone" concept must to be in the doc and doesnt in the code

**Comment by dalssoft** [comment]
Agree with @italojs that we can improve the doc with this method of cloning entities instances.

**Comment by maikmb** [comment]
Work in progress!

**Comment by jhomarolo** [comment]
Released on #66
